pwndbg/gdblib/memory.py

```diff
@@ -360,3 +360,55 @@ def update_min_addr() -> None:
     global MMAP_MIN_ADDR
     if pwndbg.gdblib.qemu.is_qemu_kernel():
         MMAP_MIN_ADDR = 0
+
+
+def fetch_struct_as_dictionary(
+    struct_name: str,
+    struct_address: int,
+    include_only_fields: list[str] | None = None,
+    exclude_fields: list[str] | None = None,
+):
+    struct_type = gdb.lookup_type("struct " + struct_name)
+    fetched_struct = poi(struct_type, struct_address)
+
+    return pack_struct_into_dictionary(fetched_struct, include_only_fields, exclude_fields)
+
+
+def pack_struct_into_dictionary(
+    fetched_struct: gdb.Value,
+    include_only_fields: list[str] | None = None,
+    exclude_fields: list[str] | None = None,
+):
+    struct_as_dictionary = {}
+
+    if include_only_fields is not None:
+        for field_name in include_only_fields:
+            key = field_name
+            value = convert_gdb_value_to_python_value(fetched_struct[field_name])
+            struct_as_dictionary[key] = value
+    else:
+        for field in fetched_struct.type.fields():
+            if field.name is None:
+                # Flatten anonymous structs/unions
+                struct_as_dictionary.update(
+                    convert_gdb_value_to_python_value(fetched_struct[field])
```

Comment:
I would add an `assert` checking `isinstance(convert_gdb_value_to_python_value(fetched_struct[field]), Dict)`, given `convert_gdb_value_to_python_value` can return other types.

Reply:
So _in theory_ it should always return a dictionary when `field.name` is `None` because only structs & unions can be anonymous. But perhaps better safe than sorry, let me know if you're happy to omit the assert or if you think it'll catch something.

Reply:
Adding the assertion was necessary for the types to work out, see the latest commits.